src/client/io_loop.rs

```diff
@@ -1565,8 +1565,18 @@ impl<T: InvokeUiSession> Remote<T> {
                     }
                 }
                 Some(message::Union::PeerInfo(pi)) => {
-                    self.handler.set_displays(&pi.displays);
-                    self.handler.set_platform_additions(&pi.platform_additions);
+                    // For follow cursor, follow window focus
+                    if pi.displays.is_empty() && pi.platform_additions.is_empty() {
+                        let config = self.handler.load_config();
+                        if config.displays_as_individual_windows != "Y"
+                            && config.show_all_displays != "Y"
```

Comment:
1. When the previous connection ended with the option show_all_displays set to "Y", the new connection only displays one option, and the follow function doesn't work until you switch display manaully. Currently it's a runtime state not a config. the recommended approach is to use a session-level variable state instead of relying on the previous connection's settings. 

2. Instead of placing the new protocol message in the PeerInfo, you can utilize a new protocol message separately to tell the client to switch display.